ebu_tt_live/node/denester.py

```diff
@@ -292,13 +325,32 @@ def recurse(div,
                     # has a dur but no begin or end times. This may well be
                     # some kind of bug or incorrect behaviour, but it's hard
                     # to sort out without breaking a bunch of stuff.
-                    p_time = c.value.computed_begin_time
-
-                    span.compBegin = span.compBegin - p_time
-                    span.compEnd = span.compEnd - p_time
-
-                    span.begin = ebuttdt.FullClockTimingType(span.compBegin)
-                    span.end = ebuttdt.FullClockTimingType(span.compEnd)
+                    p_begin_time = c.value.computed_begin_time
+                    p_end_time = c.value.computed_end_time
+
+                    if span.compBegin != p_begin_time:
+                        span.compBegin = span.compBegin - p_begin_time
+                        span.begin = ebuttdt.FullClockTimingType(
```

Comment:
Er, it doesn't. When `span.compBegin != p_begin_time` the span's `begin` is set to a `FullClockTimingType`, but otherwise it is not set. However in both cases the span's `compBegin` is recalculated. I couldn't see a neater way to avoid the duplication in lines 332 and 336.

Reply:
The _reason_ for this is that we want to avoid setting a `begin` time on a `span` if it makes no difference because the `span` begins at the same time as its parent `p`. This removes verbosity in the output so that we don't end up pushing `begin` attributes down to `span` level if we don't have to. In particular, if there is a case like:

```
<p begin="00:00:01:00"><span>Line 1</span><br/><span>Line 2</span></p>
```

we don't want to end up with:

```
<p><span begin="00:00:01:00">Line 1</span><br/><span begin="00:00:01:00">Line 2</span></p>
```

The two are subtly but importantly different. We can't have `begin` on `<br>` elements, so the resulting meaning in the latter case is that the `<br>` and the `<p>` are _always_ active but the child `<span>` elements are only active after their begin time. Whereas in the first case all the `<p>`'s children are active only after the `p` begin time.